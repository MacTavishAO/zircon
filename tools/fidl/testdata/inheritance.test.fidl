library fidl.test.json;

protocol super {
    foo(string s) -> (int64 y);
};

protocol sub {
    compose super;
};
