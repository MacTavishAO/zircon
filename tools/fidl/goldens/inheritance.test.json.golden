{
  "version": "0.0.1",
  "name": "fidl.test.json",
  "library_dependencies": [],
  "bits_declarations": [],
  "const_declarations": [],
  "enum_declarations": [],
  "experimental_resource_declarations": [],
  "interface_declarations": [
    {
      "name": "fidl.test.json/super",
      "location": {
        "filename": "inheritance.test.fidl",
        "line": 3,
        "column": 10,
        "length": 5
      },
      "methods": [
        {
          "ordinal": 5722958650322615442,
          "name": "foo",
          "location": {
            "filename": "inheritance.test.fidl",
            "line": 4,
            "column": 5,
            "length": 3
          },
          "has_request": true,
          "maybe_request": [
            {
              "type": {
                "kind": "string",
                "nullable": false
              },
              "name": "s",
              "location": {
                "filename": "inheritance.test.fidl",
                "line": 4,
                "column": 16,
                "length": 1
              },
              "field_shape_v1": {
                "offset": 16,
                "padding": 0
              }
            }
          ],
          "maybe_request_payload": "fidl.test.json/SomeLongAnonymousPrefix0",
          "maybe_request_type_shape_v1": {
            "inline_size": 32,
            "alignment": 8,
            "depth": 1,
            "max_handles": 0,
            "max_out_of_line": 4294967295,
            "has_padding": true,
            "has_flexible_envelope": false,
            "is_resource": false
          },
          "has_response": true,
          "maybe_response": [
            {
              "type": {
                "kind": "primitive",
                "subtype": "int64"
              },
              "name": "y",
              "location": {
                "filename": "inheritance.test.fidl",
                "line": 4,
                "column": 29,
                "length": 1
              },
              "field_shape_v1": {
                "offset": 16,
                "padding": 0
              }
            }
          ],
          "maybe_response_payload": "fidl.test.json/SomeLongAnonymousPrefix1",
          "maybe_response_type_shape_v1": {
            "inline_size": 24,
            "alignment": 8,
            "depth": 0,
            "max_handles": 0,
            "max_out_of_line": 0,
            "has_padding": false,
            "has_flexible_envelope": false,
            "is_resource": false
          },
          "is_composed": false
        }
      ]
    },
    {
      "name": "fidl.test.json/sub",
      "location": {
        "filename": "inheritance.test.fidl",
        "line": 7,
        "column": 10,
        "length": 3
      },
      "methods": [
        {
          "ordinal": 5722958650322615442,
          "name": "foo",
          "location": {
            "filename": "inheritance.test.fidl",
            "line": 4,
            "column": 5,
            "length": 3
          },
          "has_request": true,
          "maybe_request": [
            {
              "type": {
                "kind": "string",
                "nullable": false
              },
              "name": "s",
              "location": {
                "filename": "inheritance.test.fidl",
                "line": 4,
                "column": 16,
                "length": 1
              },
              "field_shape_v1": {
                "offset": 16,
                "padding": 0
              }
            }
          ],
          "maybe_request_payload": "fidl.test.json/SomeLongAnonymousPrefix0",
          "maybe_request_type_shape_v1": {
            "inline_size": 32,
            "alignment": 8,
            "depth": 1,
            "max_handles": 0,
            "max_out_of_line": 4294967295,
            "has_padding": true,
            "has_flexible_envelope": false,
            "is_resource": false
          },
          "has_response": true,
          "maybe_response": [
            {
              "type": {
                "kind": "primitive",
                "subtype": "int64"
              },
              "name": "y",
              "location": {
                "filename": "inheritance.test.fidl",
                "line": 4,
                "column": 29,
                "length": 1
              },
              "field_shape_v1": {
                "offset": 16,
                "padding": 0
              }
            }
          ],
          "maybe_response_payload": "fidl.test.json/SomeLongAnonymousPrefix1",
          "maybe_response_type_shape_v1": {
            "inline_size": 24,
            "alignment": 8,
            "depth": 0,
            "max_handles": 0,
            "max_out_of_line": 0,
            "has_padding": false,
            "has_flexible_envelope": false,
            "is_resource": false
          },
          "is_composed": true
        }
      ]
    }
  ],
  "service_declarations": [],
  "struct_declarations": [
    {
      "name": "fidl.test.json/SomeLongAnonymousPrefix0",
      "location": {
        "filename": "inheritance.test.fidl",
        "line": 4,
        "column": 8,
        "length": 10
      },
      "anonymous": true,
      "members": [
        {
          "type": {
            "kind": "string",
            "nullable": false
          },
          "name": "s",
          "location": {
            "filename": "inheritance.test.fidl",
            "line": 4,
            "column": 16,
            "length": 1
          },
          "field_shape_v1": {
            "offset": 0,
            "padding": 0
          }
        }
      ],
      "type_shape_v1": {
        "inline_size": 16,
        "alignment": 8,
        "depth": 1,
        "max_handles": 0,
        "max_out_of_line": 4294967295,
        "has_padding": true,
        "has_flexible_envelope": false,
        "is_resource": false
      }
    },
    {
      "name": "fidl.test.json/SomeLongAnonymousPrefix1",
      "location": {
        "filename": "inheritance.test.fidl",
        "line": 4,
        "column": 22,
        "length": 9
      },
      "anonymous": true,
      "members": [
        {
          "type": {
            "kind": "primitive",
            "subtype": "int64"
          },
          "name": "y",
          "location": {
            "filename": "inheritance.test.fidl",
            "line": 4,
            "column": 29,
            "length": 1
          },
          "field_shape_v1": {
            "offset": 0,
            "padding": 0
          }
        }
      ],
      "type_shape_v1": {
        "inline_size": 8,
        "alignment": 8,
        "depth": 0,
        "max_handles": 0,
        "max_out_of_line": 0,
        "has_padding": false,
        "has_flexible_envelope": false,
        "is_resource": false
      }
    },
    {
      "name": "fidl.test.json/SomeLongAnonymousPrefix0",
      "location": {
        "filename": "inheritance.test.fidl",
        "line": 4,
        "column": 8,
        "length": 10
      },
      "anonymous": true,
      "members": [
        {
          "type": {
            "kind": "string",
            "nullable": false
          },
          "name": "s",
          "location": {
            "filename": "inheritance.test.fidl",
            "line": 4,
            "column": 16,
            "length": 1
          },
          "field_shape_v1": {
            "offset": 0,
            "padding": 0
          }
        }
      ],
      "type_shape_v1": {
        "inline_size": 16,
        "alignment": 8,
        "depth": 1,
        "max_handles": 0,
        "max_out_of_line": 4294967295,
        "has_padding": true,
        "has_flexible_envelope": false,
        "is_resource": false
      }
    },
    {
      "name": "fidl.test.json/SomeLongAnonymousPrefix1",
      "location": {
        "filename": "inheritance.test.fidl",
        "line": 4,
        "column": 22,
        "length": 9
      },
      "anonymous": true,
      "members": [
        {
          "type": {
            "kind": "primitive",
            "subtype": "int64"
          },
          "name": "y",
          "location": {
            "filename": "inheritance.test.fidl",
            "line": 4,
            "column": 29,
            "length": 1
          },
          "field_shape_v1": {
            "offset": 0,
            "padding": 0
          }
        }
      ],
      "type_shape_v1": {
        "inline_size": 8,
        "alignment": 8,
        "depth": 0,
        "max_handles": 0,
        "max_out_of_line": 0,
        "has_padding": false,
        "has_flexible_envelope": false,
        "is_resource": false
      }
    }
  ],
  "table_declarations": [],
  "union_declarations": [],
  "type_alias_declarations": [],
  "declaration_order": [
    "fidl.test.json/super",
    "fidl.test.json/sub"
  ],
  "declarations": {
    "fidl.test.json/super": "interface",
    "fidl.test.json/sub": "interface",
    "fidl.test.json/SomeLongAnonymousPrefix0": "struct",
    "fidl.test.json/SomeLongAnonymousPrefix1": "struct"
  }
}
